File: internal/domain/theme.go
package domain

// ThemeColors holds the five user-configurable base colours for a theme.
type ThemeColors struct {
	Background  string `json:"background"`
	Surface     string `json:"surface"`
	Text        string `json:"text"`
	Accent      string `json:"accent"`
	AccentLight string `json:"accentLight"`
}

// Theme is a named colour theme that can be applied to the UI.
type Theme struct {
	ID     string      `json:"id"`
	Name   string      `json:"name"`
	Colors ThemeColors `json:"colors"`
}

File: frontend/src/types/theme.ts
export interface ThemeColors {
  background: string
  surface: string
  text: string
  accent: string
  accentLight: string
}

export interface ThemeDefinition {
  id: string
  name: string
  colors: ThemeColors
}

export const BUILT_IN_IDS = new Set(['light', 'dark'])

export const BUILT_IN_THEMES: ThemeDefinition[] = [
  {
    id: 'dark',
    name: 'Dark',
    colors: {
      background: '#0f1117',
      surface: '#161920',
      text: '#ffffff',
      accent: '#9B84D1',
      accentLight: '#C8B1F3',
    },
  },
  {
    id: 'light',
    name: 'Light',
    colors: {
      background: '#ffffff',
      surface: '#f5f3ff',
      text: '#0f1117',
      accent: '#9B84D1',
      accentLight: '#C8B1F3',
    },
  },
]
